README.md

```diff
@@ -1,69 +1,273 @@
+2. When an event occurs on an Ingress on the cluster the operator will act upon that event, depending on the type of event.
+   * **Create/Update:** The Ingress the event is created, will be filtered according the to configuration set in the ReplyURLSync config and will be synced, if it matches the filter and doesn't exist in the list of Reply URLs it will be added.
+   * **Delete:** The list of Reply URLs on the app registration will be checked and if there are any URLs that do not have an Ingress associated with it, the operator will remove the URL from the App Registration. You can change this behaviour by setting `replyURLFilter` to a regex of the URLs the operator should manage, ignoring anything that doesn't match.
+3. The operator also reconciles every 5 minutes against all Ingresses on the cluster.
 
-```sh
-kubectl apply -f config/samples/
-```
+### Azure permissions and RBAC
+Permissions needed for the operator to run properly are as follows.
 
-2. Build and push your image to the location specified by `IMG`:
-	
-```sh
-make docker-build docker-push IMG=<some-registry>/reply-urls-operator:tag
-```
-	
-3. Deploy the controller to the cluster with the image specified by `IMG`:
+#### Azure permissions
+The Operator needs to be able to read and write to the App Registrations and can be added via the `API Permissions` tab on the App Registration itself.
 
-```sh
-make deploy IMG=<some-registry>/reply-urls-operator:tag
+* API Permissions: `Application.ReadWrite.All` (Type: Application)
+
+**Note:** If you are running the cluster locally you can use the az cli to authenticate with Azure as long as your user is able to Read and Write to the App Registration that the Operator is configured to manage.
+
+#### Cluster RBAC
+
+The Operator needs the permissions below to work properly.
+
+| resources     | verbs            |
+|---------------|------------------|
+| replyurlsyncs | get, list, watch |
+| ingresses     | get, list, watch |
+
+All the RBAC files can be found in the `config/rbac` folder. They are created using markers in the Operators Go code, markers for RBAC can be found in `controllers/ingress_controller.go` and look similar to below.
+
+```go
+//+kubebuilder:rbac:groups=networking.k8s.io,resources=ingresses,verbs=get;list;watch
 ```
 
-### Uninstall CRDs
+[More information on RBAC markers](https://book.kubebuilder.io/reference/markers/rbac.html) 
+
+## Running the Operator
+
+Before deploying anything you will need an Azure App Registration that will have its Reply URLs updated by the Operator and either the same App Registration or a separate one that has the right permissions to update the App Registration's Reply URLs.
+
+[Instructions on creating an Azure App Registration](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app)
```

Comment:
ideally a script would be provided which creates the app reg and assigns the correct permissions and admin consent, but we can leave that for now given the amount of time spent on this.